src/EventService.Validation/EventUser/EditEventUserRequestValidator.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentValidation;
+using LT.DigitalOffice.Kernel.Validators;
+using LT.DigitalOffice.EventService.Models.Dto.Requests.EventUser;
+using LT.DigitalOffice.EventService.Validation.EventUser.Interfaces;
+using LT.DigitalOffice.EventService.Models.Dto.Enums;
+using LT.DigitalOffice.EventService.Data.Interfaces;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+using Microsoft.AspNetCore.JsonPatch;
+using LT.DigitalOffice.EventService.Models.Db;
+using LT.DigitalOffice.Kernel.Constants;
+using LT.DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Interfaces;
+using Microsoft.AspNetCore.Http;
+using LT.DigitalOffice.Kernel.Extensions;
+
+namespace LT.DigitalOffice.EventService.Validation.EventUser;
+
+public class EditEventUserRequestValidator : ExtendedEditRequestValidator<Guid, EditEventUserRequest>, IEditEventUserRequestValidator
+{
+  private readonly IEventUserRepository _eventUserRepository;
+  private readonly IEventRepository _eventRepository;
+  private readonly IAccessValidator _accessValidator;
+  private readonly IHttpContextAccessor _httpContextAccessor;
+
+  private void HandleInternalPropertyValidationAsync(
+    Operation<EditEventUserRequest> requestedOperation,
+    DbEvent dbEvent,
+    DbEventUser dbEventUser,
+    bool isAddEditRemoveUsers,
+    bool isUser,
+    ValidationContext<(Guid, JsonPatchDocument<EditEventUserRequest>)> context)
+  {
+    Context = context;
+    RequestedOperation = requestedOperation;
+
+    #region paths
+
+    AddСorrectPaths(
+      new List<string>
+      {
+        nameof(EditEventUserRequest.Status),
+        nameof(EditEventUserRequest.NotifyAtUtc),
+      });
+
+    AddСorrectOperations(nameof(EditEventUserRequest.Status), new List<OperationType> { OperationType.Replace });
+    AddСorrectOperations(nameof(EditEventUserRequest.NotifyAtUtc), new List<OperationType> { OperationType.Replace });
+
+    #endregion
+
+    #region Status
+
+    AddFailureForPropertyIf(
+      nameof(EditEventUserRequest.Status),
+      x => x == OperationType.Replace,
+      new Dictionary<Func<Operation<EditEventUserRequest>, bool>, string>
+      {
+        {
+          x => Enum.TryParse(typeof(EventUserStatus), x.value?.ToString(), out _) ?
+            (x.value.ToString().Trim() == "Participant" &&
+                (((dbEventUser.Status == EventUserStatus.Refused || dbEventUser.Status == EventUserStatus.Invited) && isUser)||
+                  (dbEventUser.Status == EventUserStatus.Discarded && isAddEditRemoveUsers)))||
+              (x.value.ToString().Trim() == "Refused" &&
+                  (dbEventUser.Status == EventUserStatus.Participant || dbEventUser.Status == EventUserStatus.Invited) && isUser)||
+                (x.value.ToString().Trim() == "Discarded" &&
+                  dbEventUser.Status == EventUserStatus.Participant && isAddEditRemoveUsers)
+            : false,
+            "Uncorrect user status"
```

Comment:
In my Opinion it's better to use 
`Enum.TryParse(x.value?.ToString(), out EventUserStatus status) ?
status == EventUserStatus.Participant &&`
and etc